internal/wrphandlers/mocktr181/handler.go

```
@@ -181,20 +184,34 @@ func (h Handler) get(names []string) (int64, []byte, error) {
 	return statusCode, payload, nil
 }
 
-func (h Handler) set(parameters []Parameter) int64 {
+func (h Handler) set(parameters []Parameter) (int64, []byte, error) {
+	result := Tr181Payload{}
 	for _, parameter := range parameters {
-		for _, mockParameter := range h.parameters {
+		for i := range h.parameters {
+			mockParameter := &h.parameters[i]
 			if strings.HasPrefix(mockParameter.Name, parameter.Name) {
 				if mockParameter.Access == "rw" {
```

If `mockParameter.Access` == `"r"` or something else, a set will silently fail.